contracts/Cross-chain/OmnichainGovernanceExecutor.sol

```diff
@@ -292,6 +292,7 @@ contract OmnichainGovernanceExecutor is ReentrancyGuard, BaseOmnichainController
         uint256 gasToStoreAndEmit = 30000; // enough gas to ensure we can store the payload and emit the event
 
         require(srcChainId_ == srcChainId, "OmnichainGovernanceExecutor::_blockingLzReceive: invalid source chain id");
+        bytes32 hashedPayload = keccak256(payload_);
```

Comment:
Were not saving gas setting `hashedPayload` only in the `if` statement?

Reply:
Hashing the payload outside of the if statement saves the gas used to store the failed message and emit the event if there is insufficient gas.